coffee-shop-application/src/main/kotlin/com/motycka/edu/order/OrderServiceImpl.kt

```diff
@@ -0,0 +1,121 @@
+package com.motycka.edu.order
+
+import com.motycka.edu.customer.CustomerRepository
+import com.motycka.edu.menu.InternalMenuService
+import com.motycka.edu.menu.MenuItemResponse
+import io.github.oshai.kotlinlogging.KotlinLogging
+
+private val logger = KotlinLogging.logger {}
+
+class OrderServiceImpl(
+    private val orderRepository: OrderRepository,
+    private val menuService: InternalMenuService,
+    private val customerRepository: CustomerRepository
+) : OrderService {
+
+    override suspend fun getAllOrders(): List<OrderResponse> {
+        logger.info { "Getting all orders" }
+        return orderRepository.selectAll().map { it.toOrderResponse(menuService) }
+    }
+
+    override suspend fun getOrderById(id: Long): OrderResponse {
+        logger.info { "Getting order by id: $id" }
+        return orderRepository.selectById(id)?.toOrderResponse(menuService)
+            ?: throw IllegalArgumentException("Order not found: $id")
```

Comment:
I would recommend implementing a specific exception for things like this ... it has couple advantages from ability to handle them in some common way to better searchability throughout the code.